File: src/main/scala/scala_spark/spark_book/第3章/squard_func.scala
package scala_spark.spark_book.第3章

import org.apache.spark.{SparkConf, SparkContext}

object squard_func {
  def main(args:Array[String]):Unit={
    val conf=new SparkConf().setAppName("squard_func").setMaster("local")
    val sc=new SparkContext(conf)

    val inputRDD=sc.parallelize(List(1,2,3,4))
    val resRDD1=inputRDD.map(x=>x*x)
    println(resRDD1.collect().mkString(","))

  }

}

File: src/main/java/java_spark/spark_book/第3章/squard_func.java
package java_spark.spark_book.第3章;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;

import java.util.Arrays;

public class squard_func {
    public static void main(String args[]){
        SparkConf conf=new SparkConf().setAppName("squard_func").setMaster("local");
        JavaSparkContext sc=new JavaSparkContext(conf);

        JavaRDD<Integer> inputRDD=sc.parallelize(Arrays.asList(1,2,3,4));
        JavaRDD<Integer> squardRDD=inputRDD.map(new Function<Integer, Integer>() {
            public Integer call(Integer integer) throws Exception {
                return integer*integer;
            }
        });

        System.out.println(squardRDD.collect().toString());
    }
}
